cmd/source_profile.go

```diff
@@ -0,0 +1,274 @@
+	}
+	return pg
+}
+
+type SourceProfileConnectionDynamoDB struct {
+	awsAccessKeyID     string // Same as AWS_ACCESS_KEY_ID environment variable
+	awsSecretAccessKey string // Same as AWS_SECRET_ACCESS_KEY environment variable
+	awsRegion          string // Same as AWS_REGION environment variable
+	dydbEndpoint       string // Same as DYNAMODB_ENDPOINT_OVERRIDE environment variable
+	schemaSampleSize   int64  // Number of rows to use for inferring schema (default 100,000)
+}
+
+func NewSourceProfileConnectionDynamoDB(params map[string]string) (SourceProfileConnectionDynamoDB, error) {
+	// TODO: Populate environment variables for DynamoDB if passed as flags.
+	dydb := SourceProfileConnectionDynamoDB{}
+	if awsAccessKeyID, ok := params["awsAccessKeyID"]; ok {
+		dydb.awsAccessKeyID = awsAccessKeyID
+	}
+	if awsSecretAccessKey, ok := params["awsSecretAccessKey"]; ok {
+		dydb.awsSecretAccessKey = awsSecretAccessKey
+	}
+	if awsRegion, ok := params["awsRegion"]; ok {
+		dydb.awsRegion = awsRegion
+	}
+	if dydbEndpoint, ok := params["dydbEndpoint"]; ok {
+		dydb.dydbEndpoint = dydbEndpoint
+	}
+	if schemaSampleSize, ok := params["schemaSampleSize"]; ok {
+		schemaSampleSizeInt, err := strconv.Atoi(schemaSampleSize)
+		if err != nil {
+			return dydb, fmt.Errorf("could not parse schemaSampleSize = %v as a valid int64", schemaSampleSize)
+		}
+		dydb.schemaSampleSize = int64(schemaSampleSizeInt)
+	}
+	return dydb, nil
+}
+
+type SourceProfileConnection struct {
+	ty    SourceProfileConnectionType
+	mysql SourceProfileConnectionMySQL
+	pg    SourceProfileConnectionPostgreSQL
+	dydb  SourceProfileConnectionDynamoDB
+}
+
+func NewSourceProfileConnection(source string, params map[string]string) (SourceProfileConnection, error) {
+	conn := SourceProfileConnection{}
+	switch strings.ToLower(source) {
+	case "mysql":
+		{
+			conn.ty = SourceProfileConnectionTypeMySQL
+			conn.mysql = NewSourceProfileConnectionMySQL(params)
+		}
+	case "postgresql", "postgres", "pg":
+		{
+			conn.ty = SourceProfileConnectionTypePostgreSQL
+			conn.pg = NewSourceProfileConnectionPostgreSQL(params)
+		}
+	case "dynamodb":
+		{
+			conn.ty = SourceProfileConnectionTypeDynamoDB
+			dydb, err := NewSourceProfileConnectionDynamoDB(params)
+			conn.dydb = dydb
+			if err != nil {
+				return conn, err
+			}
+		}
+	default:
+		return conn, fmt.Errorf("invalid source database: %v", source)
+	}
+	return conn, nil
+}
+
+type SourceProfileConfig struct {
+	path string
+}
+
+func NewSourceProfileConfig(path string) SourceProfileConfig {
+	return SourceProfileConfig{path: path}
+}
+
+type SourceProfile struct {
+	ty     SourceProfileType
+	file   SourceProfileFile
+	conn   SourceProfileConnection
+	config SourceProfileConfig
+}
+
+func (src SourceProfile) ToLegacyDriver(source string) (string, error) {
+	switch src.ty {
+	case SourceProfileTypeFile:
+		{
+			switch strings.ToLower(source) {
+			case "mysql":
+				return "mysqldump", nil
+			case "postgresql", "postgres", "pg":
```

Comment:
We used to default to pg if driver was not specified.
Suggest either:
1) we preserve the old default, or
2) we explicitly catch the case where source is the empty string and print a message such as: "Please specify source e.g. PostgreSQL, MySQL, DynamoDB.

Note that at the moment, if you run:
  go run github.com/cloudspannerecosystem/harbourbridge schema < examples/cart.pg_dump 

You just get:
  exit status 1

We should print an error message.

Reply:
I intended to do implemented 2. but as pointed in the other comments, seems like log.Fatalf doesn't work as intended. I have switched to using panic() to return the appropriate message.